src/lib/useEmojis.ts

```diff
@@ -9,20 +9,29 @@ import type {
 
 function reducer(state: EmojiiState, action: EmojiAction) {
   const { emoji } = action;
-  const emojiFromState = state.find((em) => em.emoji === emoji.emoji) || emoji;
+  const stateEmoji = state.find((em) => em.emoji === emoji.emoji);
+  const emojiFromState = stateEmoji || emoji;
   emojiFromState.counter = emojiFromState.counter || 0;
   switch (action.type) {
     case 'i':
       emojiFromState.counter++;
       break;
     case 'd':
       emojiFromState.counter--;
-      if (emojiFromState.counter < 0) emojiFromState.counter = 0;
+      if (emojiFromState.counter <= 0) emojiFromState.counter = 0;
```

Comment:
If the counter is already 0, there's no need to set it to 0

Reply:
correct :)